test/unit/utils/attribute-token-list.spec.js

```diff
@@ -0,0 +1,216 @@
+
+    list.add('token-1');
+    list = [...list];
+
+    assert.deepEqual(list, ['token-1'], 'Assert that tokens is "token-1".');
+  });
+
+  it('Add multiple tokenes', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1', 'token-2');
+    list = [...list];
+
+    assert.deepEqual(list, ['token-1', 'token-2'], 'Assert that tokens is "token-1" and "token-2".');
+  });
+
+  it('Just add token once', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+    list.add('token-1');
+    list = [...list];
+
+    assert.deepEqual(list, ['token-1'], 'Assert that token was only added once.');
+  });
+
+  it('Just add token once', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+    list.add('token-1');
+    list = [...list];
+
+    assert.deepEqual(list, ['token-1'], 'Assert that token was only added once.');
+  });
+
+  it('Check contains with a single token', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+
+    assert.ok(list.contains('token-1'), 'Assert that list contains token "token-1".');
+  });
+
+  it('Check contains with multiple tokens', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+    list.add('token-2');
+
+    assert.ok(list.contains('token-1'), 'Assert that list contains token "token-1".');
+  });
+
+  it('Check contains with non existing token', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+
+    assert.equal(list.contains('token-2'), false, 'Assert that list does not contains token "token-2".');
+  });
+
+  it('Check contains with non existing tokens', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+    list.add('token-2');
+
+    assert.equal(list.contains('token-3'), false, 'Assert that list does not contains token "token-3".');
+  });
+
+  it('Get item at index', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+    list.add('token-2');
+
+    assert.equal(list.item(1), 'token-2', 'Assert that "token-2" is at index 1.');
+  });
+
+  it('Get item at non existing index', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+
+    assert.equal(list.item(1), null, 'Assert that item() gives null for non existing token.');
+  });
+
+  it('Remove single token', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+    list.add('token-2');
+
+    list.remove('token-2');
+    list = [...list];
+
+    assert.deepEqual(list, ['token-1'], 'Assert that "token-2" was removed.');
+  });
+
+  it('Remove multiple tokens', function () {
+    let list = new AttributeTokenList();
+
+    list.add('token-1');
+    list.add('token-2');
+
+    list.remove('token-1', 'token-2');
+    list = [...list];
+
+    assert.deepEqual(list, [], 'Assert that "token-1" and "token-2" was removed.');
+  });
+
```

Comment:
suggestion(non-blocking): Test for removing a non-existent token

```suggestion

  it('Remove non-existent token', function () {
    let list = new AttributeTokenList();

    list.add('token-1');

    list.remove('token-3');
    list = [...list];

    assert.deepEqual(list, ['token-1'], 'Assert that token list did not change by removing non-existent token.');
  });

```